lib/Connection.js

```diff
@@ -30,68 +35,95 @@ class Connection {
    * @param {string} db
    * @param {string} user
    * @param {string} pwd
-   * @param {object} [option]
+   * @param {object} [restConfig]
    */
-  constructor(db, user, pwd, option) {
-    this.conn = {};
+  constructor(db, username, password, restConfig) {
+    // setup default values
+    this.conn = {
+      database: '*LOCAL',
+      username: '',
+      password: '',
+      xslib: 'QXMLSERV',
+      returnError: true,
+      returnXml: false,
+      transport: 'idb',
+      host: 'localhost',
+      port: 80,
+      path: '/',
+      ctl: '*here',
+      ipc: '*NA',
+      outputBuffer: 500000,
+      verbose: false,
+    };
+
     this.cmds = [];
-    this.timeout = 5000; // Default timeout is 5 seconds for sync mode.
-    this.I_DEBUG_VERBOSE = false;
-    this.conn.I_TRANSPORT_DB2_DATABASE = db; // Required Field
-    this.conn.I_TRANSPORT_DB2_USER = user; // Required Field
-    this.conn.I_TRANSPORT_DB2_PASSWORD = pwd; // Required Field
-    this.conn.I_XML_SERVICE_LIB = 'QXMLSERV'; // Default XML Service library
-    this.conn.I_TRANSPORT_CTL = '*here';
-    this.conn.I_TRANSPORT_IPC = '*NA';
-
-    if (typeof db === 'object') {
-      this.conn.I_TRANSPORT = I_TRANSPORT_DB2;
-      return;
-    }
 
-    if (option && typeof option === 'object') {
-      if (option.host) {
-        this.conn.I_TRANSPORT = I_TRANSPORT_REST;
-        this.conn.I_TRANSPORT_REST_HOST = option.host;
-        if (option.port) {
-          this.conn.I_TRANSPORT_REST_PORT = option.port;
-        } else {
-          this.conn.I_TRANSPORT_REST_PORT = 80;
-        }
-        if (option.path) {
-          this.conn.I_TRANSPORT_REST_PATH = option.path;
-        } else {
-          this.conn.I_TRANSPORT_REST_PATH = '/';
+    if (db && typeof db === 'object') {
+      // use || to fallback to default value when first argument is falsy
+      // ie (undefined, null, '', 0, false)
+      this.conn.database = db.database || this.conn.database;
+      this.conn.username = db.username || this.conn.username;
+      this.conn.password = db.password || this.conn.password;
```

Comment:
We don't allow NULL username/password?

Reply:
Should we fallback to username and password to equal `null` rather than empty string?

Reply:
I'm moving forward removing setting defaults within the Connection constructor.

That responsibility will be passed down to the individual transports.